File: src/arithmeticTriplets.ts
/**
 * Return number of unique triplets (i, j, k) where:
 * -  i < j < k
 * - nums[j] - nums[i] == diff
 * - nums[k] - nums[j] == diff
 * 
 * @param nums 
 * @param diff 
 */
function arithmeticTriplets(nums: number[], diff: number): number {
    let set = new Set();

    for (let i = 0; i < nums.length - 2; i++) {
        for (let j = i + 1; j < nums.length - 1; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                if (nums[j]! - nums[i]! == diff && nums[k]! - nums[j]! == diff) {
                    set.add([i, j, k])
                }
            }
        }
    }

    return set.size;
};

// @ts-ignore
import assert from "node:assert";
assert.strictEqual(arithmeticTriplets([0, 1, 4, 6, 7, 10], 3), 2);
assert.strictEqual(arithmeticTriplets([4, 5, 6, 7, 8, 9], 2), 2);
console.log("All test cases passed!");

File: src/arithmeticTriplets.py
class Solution:
    def arithmeticTriplets(self, nums: list[int], diff: int) -> int:
        """
        Return number of unique triplets (i, j, k) where:
        - i < j < k
        - nums[j] - nums[i] == diff
        - nums[k] - nums[j] == diff

        Time complexity is O(n) (creating set, looping over nums)
        """
        nums_set = set(nums)
        count = 0

        for num in nums:
            nums_j = diff + num
            nums_k = diff + nums_j
            if nums_j in nums_set and nums_k in nums_set:
                count += 1

        return count
